Gems/Terrain/Code/Source/TerrainRenderer/TerrainFeatureProcessor.cpp

```diff
@@ -68,76 +65,76 @@ namespace Terrain
         EnableSceneNotification();
     }
 
+    void TerrainFeatureProcessor::ConfigurePipelineState(ShaderState& shaderState, bool assertOnFail)
+    {   
+        bool success = GetParentScene()->ConfigurePipelineState(shaderState.m_shader->GetDrawListTag(), shaderState.m_pipelineStateDescriptor);
+        AZ_Assert(success || !assertOnFail, "Couldn't configure the pipeline state.");
+        if (success)
+        {
+            shaderState.m_pipelineState = shaderState.m_shader->AcquirePipelineState(shaderState.m_pipelineStateDescriptor);
+            AZ_Assert(shaderState.m_pipelineState, "Failed to acquire default pipeline state.");
+        }
+    }
+
     void TerrainFeatureProcessor::InitializeAtomStuff()
     {
         m_rhiSystem = AZ::RHI::RHISystemInterface::Get();
 
-        {
-            // Load the shader
-            constexpr const char* TerrainShaderFilePath = "Shaders/Terrain/Terrain.azshader";
-            m_shader = AZ::RPI::LoadShader(TerrainShaderFilePath);
-            if (!m_shader)
+        {       
+            auto LoadShader = [this](const char* filePath, ShaderState& shaderState)
             {
-                AZ_Error(TerrainFPName, false, "Failed to find or create a shader instance from shader asset '%s'", TerrainShaderFilePath);
-                return;
-            }
-
-            // Create the data layout
-            m_pipelineStateDescriptor = AZ::RHI::PipelineStateDescriptorForDraw{};
-            {
-                AZ::RHI::InputStreamLayoutBuilder layoutBuilder;
-
-                layoutBuilder.AddBuffer()
-                    ->Channel("POSITION", AZ::RHI::Format::R32G32_FLOAT)
-                    ->Channel("UV", AZ::RHI::Format::R32G32_FLOAT)
-                    ;
-                m_pipelineStateDescriptor.m_inputStreamLayout = layoutBuilder.End();
-            }
-
-            auto shaderVariant = m_shader->GetVariant(AZ::RPI::ShaderAsset::RootShaderVariantStableId);
-            shaderVariant.ConfigurePipelineState(m_pipelineStateDescriptor);
+                shaderState.m_shader = AZ::RPI::LoadShader(filePath);
+                if (!shaderState.m_shader)
+                {
+                    AZ_Error(TerrainFPName, false, "Failed to find or create a shader instance from shader asset '%s'", filePath);
+                    return;
+                }
```

Comment:
Just a note for the future, not necessarily this PR - ideally new systems should handle async shader loads instead of continuing to rely on sync blocking loads.

Reply:
This will be transitioning to a material that handles shader loads under the hood in a non-blocking way. I mostly wanted to just eliminate a bunch of redundant code for now.